src/uwtools/set_config.py

```diff
@@ -122,5 +122,33 @@ def create_config_obj(argv):
             out_object = config_obj
         out_object.dump_file(user_args.outfile)
 
+    ##begin values-needed
+    # if --values_needed, print a list of required configuration settings to stdout
+
+    if user_args.values_needed:
+        # first ensure all required args are present
+        assert(
+            user_args.values_needed and user_args.outfile
+        ), "args: --values_needed also requires -outfile for reference"
+
+        if outfile_type != infile_type:
+            # next, provide needed format if not performing a simple conversion
+            convert_list = [".yaml", ".yml", ".bash", ".sh", ".ini", ".IN", ".nml"]
+            if outfile_type not in convert_list:
+                print(f"""Required YAML format for field files:
+{('-' * 70)}
+sphum:
+  longname: specific humidity
+  units: kg/kg
+  profile_type: 
+    name: fixed
+    surface_value: 1.e30
+{('-' * 70)}""")
```

Comment:
I think all this type of information goes into the doc block of the FieldTableConfigure object. Then this script can call on `help()` to print that information on any of our Config objects if a user provides a flag like "show-more" or whatever makes sense.

Reply:
Ah nice, thats much better! Changing now, thanks!

Reply:
help() was a bit much, so I went with print(out_object.__doc__) which will let me add every other output object later